README.md
# spongebob : SPoNgeBOb-CAse cONveRSioN iN R

[![CRAN\_Status\_Badge](https://www.r-pkg.org/badges/version-last-release/spongebob)](https://cran.r-project.org/package=spongebob) [![CRAN\_Download\_Badge](https://cranlogs.r-pkg.org/badges/grand-total/spongebob)](https://cran.r-project.org/package=spongebob) [![Build Status](https://travis-ci.org/jayqi/spongebob.svg?branch=master)](https://travis-ci.org/jayqi/spongebob) [![AppVeyor Build Status](https://ci.appveyor.com/api/projects/status/github/jayqi/spOngEbOB?branch=master&svg=true)](https://ci.appveyor.com/project/jayqi/spOngEbOB) [![codecov](https://codecov.io/gh/jayqi/spongebob/branch/master/graph/badge.svg)](https://codecov.io/gh/jayqi/spongebob)

This R package's core is the function `tospongebob` that will convert text in R objects to [Mocking SpongeBob case](https://knowyourmeme.com/memes/mocking-spongebob). Additional features leverage it for even more fun. 

insPIreD bY A [twiTTeR CoNVErsaTIon StArtED By @jONgAllOwaY](https://twitter.com/jongalloway/status/1075889210714816512).

## Installation. inSTAllATIoN.

You can get the LateST StABle VeRsIon of `spongebob` from [CRAN](https://CRAN.R-project.org/package=spongebob):

``` r
install.packages('spongebob')
```

### Development Version. dEveLoPmEnT VerSIOn.

To install the dEveLoPmEnT VerSIOn of this package, use `remotes` to install DirEcTlY fROm gItHuB:

``` r
remotes::install_github("jayqi/spongebob")
```

## Usage. usAGe.

``` r
library(spongebob)
tospongebob(c(
    "an r package for spongebob-case conversion",
    "try to use it on anything you'd like!"
))
#> [1] "AN r pAcKaGe FOr spOnGebOb-caSe CONvERsioN"
#> [2] "tRy to USe iT on AnytHInG yOU'D lIKe!"

tospongebob(summary(datasets::chickwts))
#>      WeigHt             FeED   
#>  MIn.   :108.0   CaSein   :12  
#>  1sT qU.:204.5   HOrseBEaN:10  
#>  meDIan :258.0   LinSEeD  :12  
#>  mEan   :261.3   MeaTMeAL :11  
#>  3Rd QU.:323.5   SOYbEaN  :14  
#>  max.   :423.0   sunFLowEr:12

spongebobsay("you can even make magnificent ascii art")
#>  ----------------------------------------- 
#> | yOU CaN eVEn maKe MAgnIfICEnT ascIi ARt |
#>  ----------------------------------------- 
#>   \\
#>    \\    *
#>           *
#>      ----//-------
#>      \..C/--..--/ \   `A
#>       (@ )  ( @) \  \// |w
#>        \          \  \---/
#>         HGGGGGGG    \    /`
#>         V `---------`--'
#>             <<    <<
#>            ###   ###
```

<sup>Snippet created on 2019-03-02 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

---

<sup>The author of this package claims no ownership over SpongeBob SquarePants. SpongeBob SquarePants is the creation of Stephen Hillenburg, and its trademark is owned by Viacom International Inc. This package uses SpongeBob under fair use, and this package should only be used for personal, noncommercial purposes.</sup>


NEWS.md
# spongebob 0.3.0.9000

- `tospongebob` is now an S3 generic function! This means you can use it to convert baSIcaLLY aNyThiNg yOU CaN dREAm oF. If you dream of something that doesn't work, please [report it in our issue tracker](https://github.com/jayqi/spongebob/issues), and we'll try our best to mAKe yoUr drEaMS ComE TrUE. DOn'T LeT yOUR dreAmS juST Be DrEaMS. 

# spongebob 0.3.0

- New family of functions `spongebobsay`, `spongebobthink`, and `spongebobwhisper` that let you have ASCII Spongebob--in the style of cowsay--deliver your message WitH VisuAl flAiR aNd soPhIsTIcATIoN. 
- Standalone ASCII Mocking Spongebob art available as a character string via new function `ascii_spongebob` in case you want him to gAzE SiLEntlY aT You.

```
 ----------------------------------------- 
| neW fAMilY Of fUNCtiOns `sPOnGEboBsAy`, |
| `spONGeBoBTHinK`, aNd `sPonGEboBtHinK`  |
| THAt lET you HavE AsCiI sPOnGEbob--In   |
| the STyLe oF CowSAy--DeliVeR YoUR       |
| mesSaGe wiTh vISuaL FlAIr AND           |
| SophIStIcaTioN.                         |
 ----------------------------------------- 
  \\
   \\    *
          *
     ----//-------
     \..C/--..--/ \   `A
      (@ )  ( @) \  \// |w
       \          \  \---/
        HGGGGGGG    \    /`
        V `---------`--'
            <<    <<
           ###   ###
```

# spongebob 0.2.0

- `tospongebob`'s 1-, 2-, and 3-length subsequences now sample from a distribution whose proportions follow the golden ratio, for mORe pLeASing AEstHetiCS ThAT rEfleCt nATUre. 
- `tospongebob` now ignores non-alphabetic characters when alternating casing. This avoids situations like `"THis one"` where the whitespace ate up an uppercase subsequence. 


R/spongebob.R
#' @title spongebob : SPoNgeBOb-CAse cONveRSioN
#' @keywords internal
#' @description Convert text (and text in R objects) to
#'    \href{https://knowyourmeme.com/memes/mocking-spongebob}{Mocking SpongeBob
#'    case} and show it off in fun ways.
#'
#'    cONveRT TeXT (anD tEXt IN r obJEcTS) to
#'    \href{https://knowyourmeme.com/memes/mocking-spongebob}{mOCkinG spOngEBoB
#'    cASe} aNd shOW it OFf in FuN waYS.
#'
#' @section tospongebob -- teXt CoNVeRSIoN:
#'
#'    The core of this package is the \code{\link{tospongebob}} function. You
#'    can use it to convert character vectors! For example,
#'
#'    \preformatted{
#'        tospongebob('use it to convert character vectors!')
#'
#'        #> [1] "uSe iT to ConvERt chArACteR vECTorS!"
#'    }
#'
#'    BuT WAiT! \code{tospongebob} is actually an S3 generic function. You can
#'    use it to convert lots of things!
#'
#'    \preformatted{
#'        tospongebob(summary(datasets::chickwts))
#'
#'        #>      WeigHt             FeED
#'        #>  MIn.   :108.0   CaSein   :12
#'        #>  1sT qU.:204.5   HOrseBEaN:10
#'        #>  meDIan :258.0   LinSEeD  :12
#'        #>  mEan   :261.3   MeaTMeAL :11
#'        #>  3Rd QU.:323.5   SOYbEaN  :14
#'        #>  max.   :423.0   sunFLowEr:12
#'    }
#'
#'    Try to convert aNyThiNg yOU CaN dREAm oF. If you dream of
#'    something that doesn't work, please report it in our
#'    \href{https://github.com/jayqi/spongebob/issues}{issue tracker},
#'    and we'll try our best to mAKe yoUr drEaMS ComE TrUE. DOn'T LeT yOUR
#'    dreAmS juST Be DrEaMS.
#'
#' @section spongebobsay -- aScIi SPOnGeBOB:
#'
#'   We also have functions for getting your text spoken by an ASCII Mocking
#'   Spongebob, in the style of
#'   \href{https://en.wikipedia.org/wiki/Cowsay}{cowsay}.
#'
#'   \preformatted{
#'       spongebobsay("you can even make magnificent ascii art")
#'
#'       #>  -----------------------------------------
#'       #> | yOU CaN eVEn maKe MAgnIfICEnT ascIi ARt |
#'       #>  -----------------------------------------
#'       #>   \\
#'       #>    \\    *
#'       #>           *
#'       #>      ----//-------
#'       #>      \..C/--..--/ \   `A
#'       #>       (@ )  ( @) \  \// |w
#'       #>        \          \  \---/
#'       #>         HGGGGGGG    \    /`
#'       #>         V `---------`--'
#'       #>             <<    <<
#'       #>            ###   ###
#'   }
#'
#'   We even have several functions with DiFfEREnT SpEeCh BuBbLE StYlES! Check
#'   them out:
#'
#'   \itemize{
#'       \item{\code{\link{spongebobsay}}}
#'       \item{\code{\link{spongebobwhisper}}}
#'       \item{\code{\link{spongebobthink}}}
#'   }
#'
"_PACKAGE"
